src/runtime/SandBox.cpp

```diff
@@ -174,8 +166,8 @@ bool SandBox::createStackTrace(StackTraceDataVector& stackTraceDataVector, Execu
 
         bool alreadyExists = false;
 
-        for (size_t i = 0; i < stackTraceDataVector.size(); i++) {
-            if (stackTraceDataVector[i].first == es || stackTraceDataVector[i].first->lexicalEnvironment() == es->lexicalEnvironment()) {
+        for (size_t i = 0; i < stateStack.size(); i++) {
+            if (stateStack[i] == es || stateStack[i]->lexicalEnvironment() == es->lexicalEnvironment()) {
```

Comment:
The `stateStack` is created at line 143.
But until it get here, I couldn't find place where adding items to stateStack. 
I think I'm missing something, can you explain?

Reply:
You can find stack insertion at line 206 or 224.
FYI) To collect stack tracing information, we traverse state chain and cache state info into `stateStack`